### PR TITLE
App-project: Refactor ProjectHeader styling

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-project

## Describe your changes

The purpose of this PR is to refactor the styling of elements in ProjectHeader to fit LocaleSwitcher with any project title length. The primary change specifies a screen width of 1200px as the breakpoint for when ProjectHeader should show a nav dropdown and stack its elements in a column rather than a row.

I also fixed some long standing styling bugs - ApprovedIcon retains its circle shape, and the project page navigation links don't overlap each other as screen width decreases.

I attached three screen recording examples for review. Project titles long enough to become multi-line create an issue where the containing `<anchor>` element will not size-down to fit exactly to its content size. I tried multiple css solutions, but concluded that in the case of multi-line titles, there's always going to be a gap in between the title and the approved checkmark with our current page design.

Example of short project title (TESS) with locale dropdown:

https://user-images.githubusercontent.com/23665803/179761166-696cfa5e-f677-41fd-acdf-bea62d56e858.mov

Example of long project title with locale dropdown. Note that the border around the title is simply to illustrate the issue mentioned above:

https://user-images.githubusercontent.com/23665803/179761458-f858321f-6373-4bd6-9afe-5495032fd397.mov

Example of long project title without locale dropdown:

https://user-images.githubusercontent.com/23665803/179761622-7b5b8ab4-8137-408c-98c1-f902e662e55e.mov

## How to Review

Compare the attached videos with the following production links
- TESS: https://frontend.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess
- Notes From Nature with long title: https://frontend.preview.zooniverse.org/projects/md68135/notes-from-nature-digitizing-biological-collections-in-canada

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by seanmiller26** [comment]
Really appreciate the cleanup, thanks! The oval checkmark and links running into each other needed attention. 
I think we may want to consider a medium viewpoint where the collapsed header and language selection are next to the project title, in a single line. Could be helpful for phablets, tablets and/or landscape mobile view. Otherwise, looks good!

**Comment by goplayoutside3** [comment]
> I think we may want to consider a medium viewpoint where the collapsed header and language selection are next to the project title, in a single line. Could be helpful for phablets, tablets and/or landscape mobile view. Otherwise, looks good!

Good point! I think in practice, that type of layout (title + language + dropdown navigation in one row) will only fit on the largest of ipad screens, but still worth trying. I'll upload a new screen recording with a demo.

**Comment by goplayoutside3** [comment]
@seanmiller26 Here's two new screen recordings. The "approved" checkmark is still a bit awkward, but please let me know if this styling looks okay for launching the language dropdown menu.


https://user-images.githubusercontent.com/23665803/180232294-f525ee73-6bc2-4842-88f3-1cca22238445.mp4


https://user-images.githubusercontent.com/23665803/180232380-1fd4c436-bd60-48fc-85db-e986fa5b47b8.mp4

**Comment by shaunanoordin** [comment]
Tagging in for a review! One sec while I pull the files to local

File: packages/app-project/src/components/ProjectHeader/ProjectHeader.js [hunk 1]
```diff
@@ -15,49 +15,73 @@ import UnderReviewLabel from './components/UnderReviewLabel'
...
+  const showDropdownWithRow = (hasTranslations && width < 1200) || (!hasTranslations && width < 1100)
+  const showDropdownWithColumn = (hasTranslations && width < 1000) || (!hasTranslations && width < 900)
+
+  const calcMaxWidth = () => {
+    if (hasTranslations && width >= 1200) {
+      return '560px'
+    } else {
+      return '600px'
+    }
+  }
```
Comment (shaunanoordin) [inline packages/app-project/src/components/ProjectHeader/ProjectHeader.js:39]: [documentation] / [coding style] This section makes perfect sense logically, but I'm mildly concerned that there are a bunch of magic numbers floating around here, which may go unexplained to somebody new who's looking at the code but has never seen this PR.

Recommended actions:
- By default, keep it as is, this isn't a major problem for this PR.
- However, if you can, try to get a second dev opinion to see if they think organising the numbers into something like `const HEADER_WIDTH_FOR_DESKTOP = 1200` is more readable to them.
- Or maybe just add some comments, like `/* Hey we visually tested this and the perfect breakpoint is at width 1200, but be sure to revise this if our general page CSS starts shifting to different breakpoints */`

File: packages/app-project/src/components/ProjectHeader/ProjectHeader.js [hunk 2]
```diff
@@ -67,13 +91,15 @@ ProjectHeader.propTypes = {
...
-    href: string,
-    text: string
-  })),
+  navLinks: arrayOf(
+    shape({
+      href: string,
+      text: string
+    })
+  ),
   screenSize: string,
```
Comment (shaunanoordin) [inline packages/app-project/src/components/ProjectHeader/ProjectHeader.js:100]: Begone! You are not wanted here no more, screenSize!

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `app-project`

This PR refactors the project header on FEM project pages (example: Planet Hunters Tess)](https://frontend.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess)

- No behavioural changes noted ; only visual/layout.
  - Project Header looks good when there's a Locale Switcher (language dropdown) component
  - ...or when there's an "approved" ✔️ icon
  - ...or when it has a long project name (causing the project name to span multiple text lines)
  - ...or any combination of the above
- ProjectHeader now uses `react-resize-detector` for determining its screen size, instead of being explicitly passed a `screenSize` value

Code read looks fine, and testing raised no issues. My only notable comment is the presence of magic numbers in the code (e.g. `width < 1200`), whose context or intent may be lost if another dev isn't looking at the code through the lens of this PR. Some inline comments in there may help.

### Testing

Tested on localhost with macOS + Chrome / FF / Safari (minimal testing with Safari, though)

Setup:
- Select a test project with translations AND has an "approved" icon, e.g. [Planet Hunters TESS](https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess?env=production)
- For comparison, also check a test project without either of those, e.g. [Test Project 2022](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022?env=staging)
- We're only testing the project _home_ page, though the project header _should_ be the same across all project pages (e.g. Classify page, Project About page, etc)

Testing Steps:
- Load the home page and observe visual layout of the project header
- Resize the page to check different width options. (i.e. test for different devices)
- Expectations:
  - Project title should always be readable within the visible screen space
  - If there's a Locale Switcher or "approved" icon, they should also be within the visible screen space.

Testing Scenarios:
- Project with Locale Switcher vs without.
  - Ditto with "approved" icon vs without
- Project with long name (i.e. spans multiple text lines) vs short names
  - Can be done via editing the text within the dev console.
- Random combination of the above

Visual tests look fine across all three browsers 👍 

### Dev Notes

Additional minor comments! Please acknowledge or ignore as you see fit:

- The Locale Switcher seems to be working great (on the development environment), so are we making it live already? I'm asking because I noticed there's an `xit`-ed test in ProjectHeader.spec.js that would have specifically tested for this, and the component seems as ready as it'll ever be.
- Why doesn't Project Header have its own Storybook story? I mean, it's a job for a separate PR, but it definitely wouldn't been pretty neat to have been able to toggle on/off the "approved" icons and test for long names and etc.

### Status

LGTM! 👍

File: packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js [hunk 3]
```diff
@@ -62,7 +63,7 @@ function DropdownNav({
     >
       <Box
         as='ul'
-        pad="none"
+        pad='0px'
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js:66]: 'none' is not a real css padding value, so I think Grommet was managing to make it look like 0px. Now it's clear that any inherited padding is set to a value of 0.

File: packages/app-project/src/components/ProjectHeader/ProjectHeader.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-project/src/components/ProjectHeader/ProjectHeader.js]: I'll add a note! From the design_thoughts Slack channel, there may be further restyling of ApprovedIcon's placement.

**Comment by goplayoutside3** [comment]
> The Locale Switcher seems to be working great (on the development environment), so are we making it live already? I'm asking because I noticed there's an xit-ed test in ProjectHeader.spec.js that would have specifically tested for this, and the component seems as ready as it'll ever be.

PR #3469 is the one for enabling LocaleSwitcher on the live site.

File: packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js:66]: `0px` isn’t a valid CSS length either. Minor change but it should just be `0`.

[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js:66]: Oh shoot you're right!

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/app-project/src/components/ProjectHeader/components/DropdownNav/DropdownNav.js:66]: I think browsers will accept it but there might be a warning in dev tools.

**Pull request merged**